DEV_GUIDE.md

```diff
@@ -442,3 +442,18 @@ and one repository [secret](https://docs.github.com/en/actions/security-guides/u
 * `REGISTRY_TOKEN` secret - the access token that corresponds to `REGISTRY_USERNAME` 
 
 The workflow will push the container image to `${REGISTRY_DESTINATION}` so ensure that the `${REGISTRY_USERNAME}` user has sufficient write privileges. 
+
+# Deprecation Policy
+
+We want to let users know about upcoming changes to APIs and give them sufficient time to adapt.
+
+When there is an API deprecation, it must be announced in the [CHANGELOG](./CHANGELOG.md) of the coming release under
+a section title "Changes, deprecations and removals".
+
+Deprecated features become eligible for removal in the third minor release made following the release with the 
+deprecation announcement.  There is an additional condition that at least three months must have elapsed too.  When
```

Comment:
I don't like the time condition. I don't think it adds anything useful for users if we are following semver properly.

Thinking about semver this policy only really works in `0.y.z` time frame (which is fine) once we hit `1.0.0` deleting the deprecated things will have to happen with a major bump.

Reply:
> Thinking about semver this policy only really works in 0.y.z time frame (which is fine) once we hit 1.0.0 deleting the deprecated things will have to happen with a major bump.

Yes I agree, once we reach 1.0.0 this will need changed.

> I don't like the time condition. I don't think it adds anything useful for users if we are following semver properly.

I don't feel strongly but others seemed to care.  I'm happy to go with the consensus.